salt/_modules/metalk8s_network.py

```diff
@@ -0,0 +1,44 @@
+'''Metalk8s network related utilities.'''
+
+import ipaddress
+import logging
+
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'metalk8s_network'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def pick_nth_ip(service_network, index):
```

Comment:
As @slaperche-scality suggested, it makes sense to define this as a _private_ helper function, and define specific methods for `get_kubernetes_service_ip` and `get_cluster_dns_ip`.

The rationale is simple: even if `pick_nth_ip` is an explicit name w.r.t what it does, it doesn't tell us **why** we want to pick the 10th for DNS, nor the first for the `kubernetes` service.

So, this module could instead look like:

```
# imports and other init stuff...

def _pick_nth_ip(network_range, n):
    """Pick the nth IP address from a network range, provided in CIDR notation."""
   # Same code...
   return ip

def get_cluster_dns_ip():
    """Return the CoreDNS cluster IP based on pillar configuration.

    This IP is arbitrarily selected as the tenth IP from the Service network range.
    """
    return _pick_nth_ip(__pillar__['networks.service'], 10)


def get_kubernetes_service_ip():
    """Return the Kubernetes service cluster IP based on pillar configuration.

    This IP is arbitrarily selected as the second IP from the Service network range.
    """
    return _pick_nth_ip(__pillar__['networks.service'], 2)
```

It yields that, in SLS files, no one will wonder "_why do we pick these values?!_", and instead will refer to these methods for understanding. You could have achieved the same result by adding a comment in those SLS files, but since these IPs are used several times, you gain the benefit of de-duplication.